osquery/tables/system/windows/shimcache.cpp

```diff
@@ -0,0 +1,216 @@
+          // values and if the Shimcache delimiter exists at the specific
+          // substring
+          if ((boost::starts_with(data, WIN8_START)) &&
+              (data.substr(WIN8, 8) == WIN_81_10_SHIMCACHE_DELIMITER)) {
+            execution_flag_exists = "true";
+            delimter = WIN_81_10_SHIMCACHE_DELIMITER;
+          } else if (boost::starts_with(data, WIN10_START) &&
+                     (data.substr(WIN10_PRE_CREATOR, 8) ==
+                      WIN_81_10_SHIMCACHE_DELIMITER)) {
+            delimter = WIN_81_10_SHIMCACHE_DELIMITER;
+          } else if (boost::starts_with(data, WIN10_CREATOR_START) &&
+                     (data.substr(WIN10_CREATOR, 8) ==
+                      WIN_81_10_SHIMCACHE_DELIMITER)) {
+            delimter = WIN_81_10_SHIMCACHE_DELIMITER;
+          } else {
+            LOG(WARNING) << "Unknown or unsupported shimcache data!";
+            LOG(WARNING) << data.substr(256, 8);
+            results = {};
+            return results;
+          }
+          bool first_run = true;
+
+          // Find all entries base on shimcache data delimter
+          while ((pos = data.find(delimter)) != std::string::npos) {
+            token = data.substr(0, pos);
+
+            // Skip all the data before the first delimter match
+            if (token.length() > 20) {
+              if (first_run) {
+                first_run = false;
+                data.erase(0, pos + delimter.length());
+                continue;
+              }
+              auto shimcache_results = parseShimcacheData(
+                  data, pos, token, execution_flag_exists, delimter);
+              r["entry"] = INTEGER(entry);
+              r["path"] = TEXT(shimcache_results.path);
+              r["modified_time"] = INTEGER(shimcache_results.last_modified);
+              r["execution_flag"] = TEXT(shimcache_results.execution_flag);
+              results.push_back(r);
+              entry++;
+            } else {
+              LOG(ERROR)
+                  << "ERROR: Shimcache entry does not meet length requirements";
+              LOG(ERROR) << token;
+            }
+            data.erase(0, pos + delimter.length());
+          }
+
+          // Get last appcopmat entry
+          token = data.substr(0, pos);
+          auto shimcache_results = parseShimcacheData(
+              data, pos, token, execution_flag_exists, delimter);
+          r["entry"] = INTEGER(entry);
+          r["path"] = TEXT(shimcache_results.path);
+          r["modified_time"] = INTEGER(shimcache_results.last_modified);
+          r["execution_flag"] = TEXT(shimcache_results.execution_flag);
+          results.push_back(r);
+        }
+      }
+    }
+  }
+  return results;
+}
+
+QueryData genShimcache(QueryContext& context) {
+  QueryData results;
+  std::set<std::string> shimcacheResults;
+  Row r;
+
+  expandRegistryGlobs(SHIMCACHE_CONTROLSET, shimcacheResults);
+  results = parseRegistry(shimcacheResults, r, results);
```

Comment:
I would change this slightly, to take the results by reference, return `void` and not supply r.

so it becomes:
```
parseRegistry(shimcacheResults, results);
```

Reply:
Fixed